### PR TITLE
Implement `blas_get_num_threads`

### PR DESCRIPTION
**Pull request by eschnett**
ref #10028

### CODE REVIEW

File: base/linalg/blas.jl [hunk 1]
```diff
@@ -65,6 +65,13 @@ Set the number of threads the BLAS library should use.
 """
 blas_set_num_threads
 
+"""
+    blas_get_num_threads()
```
Comment (tkelman) [inline base/linalg/blas.jl]: add to rst docs too

**Comment by tkelman** [comment]
This will cause openblas bindings to fail for all versions prior to that API being introduced. It should be conditional on openblas version.

**Comment by eschnett** [comment]
I just realize that the return value of `blas_get_num_threads` is not useful. It is in particular not the number that is set via `blas_set_num_threads`.

**Comment by kitkat2233** [comment]
@eschnett can you elaborate on that? For me it's returning the value set. Which OS and BLAS vendor are you using?

**Comment by kitkat2233** [comment]
Oh I am seeing this result on travis.  It must depend on the system.

**Pull request closed without merging**